pkg/scheduler/framework/status.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package framework
+
+import "strings"
+
+// StatusCode is the status code of a Status, returned by a plugin.
+type StatusCode int
+
+// Pre-defined status codes.
+const (
+	// Success signals that a plugin has completed its run successfully.
+	// Note that a nil *Status is also considered as a Success Status.
+	Success StatusCode = iota
+	// internalError signals that a plugin has encountered an internal error.
+	// Note that this status code is NOT exported; to return an internalError status, use the
+	// FromError() call.
+	internalError
```

Comment:
I wonder why is this internal state  in the middle of all the states?

Reply:
Hi Ryan! This is not an internal state per se, but to indicate that a plugin could not complete its task due to an error. And I couldn't use `error` due to the name collision :(